src/functions/indAffineDirect.jl

```diff
@@ -0,0 +1,105 @@
+### CONCRETE TYPE: DIRECT PROX EVALUATION
+# prox! is computed using a QR factorization of A'.
+
+struct IndAffineDirect{R <: Real, T <: RealOrComplex{R}, M <: AbstractMatrix{T}, V <: AbstractVector{T}, F <: Factorization} <: IndAffine
```

Comment:
Am I right If I say that it will fail to create an object on input
```julia
A = rand(m, n) + im * rand(m, n)
b = zeros(m)
``` 
for any arbitrary `m,n` that is acceptable. 

I think it's not okay, but maybe there's a reasonable workaround without adding `T2`.

Testcase:
```julia
const RorC = Union{Real, Complex}

function testt(a::T, b::T) where {T<: RorC}
  a + b
end

testt(2., 3.)
testt(2. + im * 3., 3. + im * 2.)
testt(2. + im * 3., 3.) #Fails
```